### PR TITLE
[OmniBridge]: Support bind token

### PR DESCRIPTION
**Pull request by olga24912**
Support the ability to call bind token in NEP141-locker. Added the ability to extract proof by tx_hash and log_index_in_receipt for convenience in tests.

* https://github.com/Near-One/rainbow-bridge-sdk-rs/issues/31

### CODE REVIEW

**Comment by kiseln** [comment]
I have code stashed somewhere that allows us to lookup the event based on the signature (had to drop the work due to other urgent tasks). I suggest we bring it back and get rid of supplying indexes altogether

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs [hunk 1]
```diff
@@ -424,6 +426,66 @@ impl OmniConnector {
         Ok(outcome)
     }
 
+    pub async fn bind_token_eth_evm_prover(
```
Comment (kiseln) [inline bridge-sdk/connectors/omni-connector/src/omni_connector.rs:429]: Not sure I understand the naming of this method

**Review by kiseln — CHANGES_REQUESTED** [review]

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs (hunk 1 above)
Comment (olga24912) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs]: fixed: https://github.com/Near-One/rainbow-bridge-sdk-rs/pull/37/commits/18fb13334bb0ada4358063145d7ae121efd9d5b8

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs [hunk 2]
```diff
@@ -421,6 +423,61 @@ impl OmniConnector {
         Ok(outcome)
     }
 
+    pub async fn bind_token_with_evm_prover(&self, tx_hash: TxHash) -> Result<CryptoHash> {
+        let eth_endpoint = self.eth_endpoint()?;
+
+        // keccak(DeployToken(address,string,string,string,uint8))
+        let event_topic =
+            H256::from_str("0x47f94ffff8bf287abb76c2b671306c7cb14769c4652a1d2fa447722e82107719")
```
Comment (frolvanya) [inline bridge-sdk/connectors/omni-connector/src/omni_connector.rs:431]: @olga24912 Why can't we compute this hash dynamically?

Comment (kiseln) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs:431]: It's how I started it in other connectors. We can introduce a keccak dependency and calculate it, I just haven't found time for it yet

Comment (frolvanya) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs:431]: I can create a refactor PR for that today. Will it be a useful change or it's better to ignore it for now?

Comment (kiseln) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs:431]: Would be nice

Comment (kiseln) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs]: Let's remove `log_index` from bind token command and merge this

**Review by kiseln — APPROVED** [review]

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs (hunk 2 above)
Comment (kiseln) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs]: I did that but now someone else needs to approve @frolvanya

**Review by frolvanya — APPROVED** [review]

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs (hunk 2 above)
Comment (frolvanya) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs]: Done, but we have a conflict in Cargo.lock file. Can you please resolve it?

**Review by frolvanya — APPROVED** [review]

**Pull request merged**